src/CarsModel.js

```diff
@@ -0,0 +1,47 @@
+import { MissionUtils } from "@woowacourse/mission-utils";
+import RULES from "./constants/rule.js";
+
+class CarsModel {
+  #carsMap;
+
+  constructor(input) {
+    const cars = input.split(",");
+    this.#carsMap = new Map(cars.map((car) => [car.trim(), 0]));
+  }
+
+  getCarNames() {
+    return [...this.#carsMap.keys()];
+  }
+
+  getCarsMap() {
+    return this.#carsMap;
+  }
+
+  #isCarMove() {
+    const randomNum = MissionUtils.Random.pickNumberInRange(
+      RULES.RANDOM_MIN,
+      RULES.RANDOM_MAX,
+    );
+
+    return randomNum >= RULES.MOVE_THRESHOLD;
+  }
+
+  moveCars() {
+    this.#carsMap.forEach((carMoveCount, carName) => {
+      if (this.#isCarMove()) this.#carsMap.set(carName, carMoveCount + 1);
+    });
+  }
+
+  getWinners() {
+    const winningMoveCount = Math.max(...this.#carsMap.values());
+
+    const winners = [];
+    this.#carsMap.forEach((carMoveCount, carName) => {
```

Comment:
Javascript에 내장되어 있는 filter 메서드를 사용하면 winners 배열을 새로 만들어서 push를 하지 않아도 되지 않을까요?
이번 1주차 피드백에 Javascript 내장 함수를 적극 활용하라 라는 피드백이 있어서 말씀드립니다!

Reply:
저두 이렇게 할까 `filter`를 사용할까 고민을 많이 했는데 지적해주셔서 정말 감사드려요! ㅎㅎ
고민 끝에 `push`를 한 이유는 아무래도 `Map` 객체라서 인것 같아요... `filter`를 사용하게 되면 오히려 로직이 복잡해져서.. 가독성이 조금 떨어진다고 생각했거든요 ㅠㅠ 
```suggestion
  const winners = Array.from(this.#carsMap)
    .filter(([_, carMoveCount]) => carMoveCount === winningMoveCount)
    .map(([carName]) => carName);
```